qdev_wrappers/sweep_functions.py

```diff
@@ -83,18 +83,30 @@ def _do_measurement_single(measurement: Measure, meas_params: tuple,
             # Ensure the correct scaling before saving
             plot.autorange()
             plot.save()
-            if 'pdf_subfolder' in CURRENT_EXPERIMENT:
+
+
+            if any(k in CURRENT_EXPERIMENT for k in ('pdf_subfolder', 'png_subfolder')):
                 plt.ioff()
                 pdfplot, num_subplots = _plot_setup(data, meas_params, useQT=False)
                 # pad a bit more to prevent overlap between
                 # suptitle and title
                 pdfplot.rescale_axis()
                 pdfplot.fig.tight_layout(pad=3)
-                title_list = plot.get_default_title().split(sep)
-                title_list.insert(-1, CURRENT_EXPERIMENT['pdf_subfolder'])
-                title = sep.join(title_list)
 
-                pdfplot.save("{}.pdf".format(title))
+                if 'pdf_subfolder' in CURRENT_EXPERIMENT:
+                    title_list = plot.get_default_title().split(sep)
+                    title_list.insert(-1, CURRENT_EXPERIMENT['pdf_subfolder'])
+                    title = sep.join(title_list)
+                    pdfplot.save("{}.pdf".format(title))
+
+                if 'png_subfolder' in CURRENT_EXPERIMENT:
+                    # Hack to save PNG also
+                    title_list_png = plot.get_default_title().split(sep)
+                    title_list_png.insert(-1, CURRENT_EXPERIMENT['png_subfolder'])
+                    title_png = sep.join(title_list_png)
+
+                    plt.savefig("{}.png".format(title_png),dpi=500)
+
```

Comment:
I would do this as (untested), This should change below in doND too

```
if 'pdf_subfolder' in CURRENT_EXPERIMENT or 'png_subfolder' in CURRENT_EXPERIMENT:
    title_list = plot.get_default_title().split(sep)
    title_list.insert(-1, CURRENT_EXPERIMENT['pdf_subfolder'])
    title = sep.join(title_list)
    if 'pdf_subfolder' in CURRENT_EXPERIMENT
         pdfplot.save("{}.pdf".format(title))
   if 'png_subfolder' in CURRENT_EXPERIMENT:
        pdfplot.save("{}.pdf".format(title))
``` 

And then change the pdfplot variable to something more descriptive